src/EventSubscriber/WebSSOEventSubscriber.php

```diff
@@ -0,0 +1,312 @@
+            'reach_api_rt' => $user->hasAccessToApiRealTime() ? 1 : 0
+        ];
+        $centreonSession = new \Centreon($sessionUserInfos);
+        $request->getSession()->start();
+        $request->getSession()->set('centreon', $centreonSession);
+        $_SESSION['centreon'] = $centreonSession;
+    }
+
+    /**
+     * Create token if not exist
+     *
+     * @param string $sessionId
+     * @param integer $webSSOConfigurationId
+     * @param Contact $user
+     * @param string $clientIp
+     */
+    private function createTokenIfNotExist(
+        string $sessionId,
+        int $webSSOConfigurationId,
+        Contact $user,
+        string $clientIp
+    ): void {
+        $authenticationTokens = $this->authenticationService->findAuthenticationTokensByToken(
+            $sessionId
+        );
+        if ($authenticationTokens === null) {
+            $sessionExpireOption = $this->optionService->findSelectedOptions(['session_expire']);
+            if (!empty($sessionExpireOption)) {
+                $this->sessionExpirationDelay = (int) $sessionExpireOption[0]->getValue();
+            }
+            $token = new ProviderToken(
+                $webSSOConfigurationId,
+                $sessionId,
+                new \DateTime(),
+                (new \DateTime())->add(new \DateInterval('PT' . $this->sessionExpirationDelay . 'M'))
+            );
+            $this->createAuthenticationTokens(
+                $sessionId,
+                $user,
+                $token,
+                null,
+                $clientIp,
+            );
+        }
+    }
+
+    /**
+     * create Authentication tokens
+     *
+     * @param string $sessionToken
+     * @param ContactInterface $contact
+     * @param ProviderToken $providerToken
+     * @param ProviderToken|null $providerRefreshToken
+     * @param string|null $clientIp
+     * @throws AuthenticationException
+     */
+    private function createAuthenticationTokens(
+        string $sessionToken,
+        ContactInterface $contact,
+        ProviderToken $providerToken,
+        ?ProviderToken $providerRefreshToken,
+        ?string $clientIp,
+    ): void {
+        $isAlreadyInTransaction = $this->dataStorageEngine->isAlreadyinTransaction();
+
+        if (!$isAlreadyInTransaction) {
+            $this->dataStorageEngine->startTransaction();
+        }
+        try {
+            $session = new Session($sessionToken, $contact->getId(), $clientIp);
+            $this->sessionRepository->addSession($session);
+            $this->authenticationRepository->addAuthenticationTokens(
+                $sessionToken,
+                $providerToken->getId(),
+                $contact->getId(),
+                $providerToken,
+                $providerRefreshToken
+            );
+            if (!$isAlreadyInTransaction) {
+                $this->dataStorageEngine->commitTransaction();
+            }
+        } catch (\Exception) {
```

Comment:
Exception is lost, you should at least log it

Reply:
good point, i forgot to add log in all the flow